stix_shifter_modules/trendmicro_vision_one/stix_translation/query_constructor.py

```diff
@@ -73,34 +73,58 @@ def _parse_reference(self, stix_field, value_type, mapped_field, value, comparat
             return "{mapped_field} {comparator} {value}".format(
                 mapped_field=remove_enclosed_quote(mapped_field), comparator=comparator, value=value)
 
-    def _parse_mapped_fields(self, expression, value, comparator, stix_field, mapped_fields_array):
-        comparison_string = ""
-        is_reference_value = self._is_reference_value(stix_field)
-        value_type = self._check_value_type(expression.value) if is_reference_value else None
-        mapped_fields_count = len(mapped_fields_array)
-
-        for mapped_field in mapped_fields_array:
-            mapped_field = remove_enclosed_quote(mapped_field)
-            if is_reference_value:
-                parsed_reference = self._parse_reference(stix_field, value_type, mapped_field, value, comparator)
-                if not parsed_reference:
-                    continue
-                comparison_string += parsed_reference
-            else:
-                if comparator == ":":
-                    comparison_string += "{mapped_field}{comparator}{value}".format(mapped_field=mapped_field, comparator=comparator, value=value)
-                else:
-                    comparison_string += "{mapped_field} {comparator} {value}".format(mapped_field=mapped_field, comparator=comparator, value=value)
+    @staticmethod
+    def _check_value_type(value):
+        """
+        Determine the type (ipv4, ipv6, mac, date, etc) of the provided value.
+        See: https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_utils/stix_translation/src/json_to_stix/observable.py#L1
+        :param value: query value
+        :type value: int/str
+        :return: type of value
+        :rtype: str
+        """
+        value = str(value)
+        for key, pattern in observable.REGEX.items():
+            if bool(re.search(pattern, value)):
+                return key
+        return None
 
-            if mapped_fields_count > 1:
-                comparison_string += " OR "
-                mapped_fields_count -= 1
-        return comparison_string
+    def _parse_mapped_fields(self, value, comparator, mapped_fields_array) -> str:
+        """Convert a list of mapped fields into a query string."""
+        comparison_strings = []
+        value_type = None
+        str_ = None
+
+        if isinstance(value, str):
+            value = [value]
+
+        for val in value:
+            value_type = self._check_value_type(val)
+
+            for mapped_field in mapped_fields_array:
+                # Only use the ipv4 fields when the value is an actual ipv4 address or range
+                skip = ('ipv4' in mapped_field and value_type not in ['ipv4', 'ipv4_cidr'])
+                # Only use the ipv6 fields when the value is an actual ipv6 address or range
+                skip = skip or ('ipv6' in mapped_field and value_type not in ['ipv6', 'ipv6_cidr'])
```

Comment:
I know this check exist in some other connectors. I wonder if these IP address checking necessay.

Reply:
thanks, removed this skip check belonging to another connector